tests/installation/welcome.pm

```diff
@@ -18,36 +18,49 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    my @welcome_tags = [qw(inst-welcome inst-welcome-confirm-self-update-server)];
-    ensure_fullscreen;
-    my $bootup_timeout = 500;
-    if (get_var("BETA")) {
-        assert_screen "inst-betawarning", $bootup_timeout;
-        send_key "ret";
-        assert_screen @welcome_tags, 10;
+    while (1) {
+        my @welcome_tags = ('inst-welcome', 'inst-welcome-confirm-self-update-server', 'scc-invalid-url');
+        # replace tag 'inst-welcome' with 'inst-betawarning' to avoid race condition
+        shift @welcome_tags && push @welcome_tags, 'inst-betawarning' if get_var('BETA');
```

Comment:
I would prefer that too - the proposed form requires quite some knowledge about the preference of perl operators